src/pages/ReimbursementAccount/BankInfo/BankInfo.tsx

```diff
@@ -60,20 +55,18 @@ function BankInfo({reimbursementAccount, reimbursementAccountDraft, plaidLinkTok
     const submit = useCallback(() => {
         if (setupType === CONST.BANK_ACCOUNT.SETUP_TYPE.MANUAL) {
             BankAccounts.connectBankAccountManually(
-                Number(getDefaultValueForReimbursementAccountField(reimbursementAccount, bankInfoStepKeys.BANK_ACCOUNT_ID, '0')),
+                Number(reimbursementAccount?.achData?.bankAccountID ?? '0'),
                 values[bankInfoStepKeys.ACCOUNT_NUMBER],
                 values[bankInfoStepKeys.ROUTING_NUMBER],
                 values[bankInfoStepKeys.PLAID_MASK],
             );
         } else if (setupType === CONST.BANK_ACCOUNT.SETUP_TYPE.PLAID) {
-            BankAccounts.connectBankAccountWithPlaid(Number(getDefaultValueForReimbursementAccountField(reimbursementAccount, bankInfoStepKeys.BANK_ACCOUNT_ID, '0')), {
-                [bankInfoStepKeys.ROUTING_NUMBER]: values[bankInfoStepKeys.ROUTING_NUMBER],
-                [bankInfoStepKeys.ACCOUNT_NUMBER]: values[bankInfoStepKeys.ACCOUNT_NUMBER],
-                [bankInfoStepKeys.PLAID_MASK]: values[bankInfoStepKeys.PLAID_MASK],
-                [bankInfoStepKeys.IS_SAVINGS]: values[bankInfoStepKeys.IS_SAVINGS],
-                [bankInfoStepKeys.BANK_NAME]: values[bankInfoStepKeys.BANK_NAME],
-                [bankInfoStepKeys.PLAID_ACCOUNT_ID]: values[bankInfoStepKeys.PLAID_ACCOUNT_ID],
-                [bankInfoStepKeys.PLAID_ACCESS_TOKEN]: values[bankInfoStepKeys.PLAID_ACCESS_TOKEN],
+            BankAccounts.connectBankAccountWithPlaid(Number(reimbursementAccount?.achData?.bankAccountID ?? '0'), {
+                [bankInfoStepKeys.ROUTING_NUMBER]: values[bankInfoStepKeys.ROUTING_NUMBER] ?? '',
```

Comment:
I think we shouldn't use variables as a key for objects in typescript. TS gives us confidence about object shape, so don't need to use variables to get data from particular attributes.
Please type values and use e.g. `values.routingNumber` instead current way.
Please review all files to change this approach.

Reply:
Using the variables as key can be useful in cases where the keys are dynamic or cases where we need to do a change in the keys, we only change the value in CONST and there is no need to change the other places. For me there is no problem in using the variables for keys (we have examples in the code where this is necessary btw), and there are no rule enforcements for that, so it's up to you @MrMuzyk 🙂

Reply:
The original idea behind it was to avoid changes in multiple places in case any of the keys would change. Given that its suppose to be just TS migration Id leave it as it is for now

Reply:
If we base on same type Your idea is safe and this is more TS way than the current solution.

Reply:
What bothers me is in the variable used as a key You can put anything, also a string that is not an attribute of the object. Su usage of data in that way is insecure.